code/datums/diseases/advance/symptoms/wizarditis.dm

```diff
@@ -100,10 +100,37 @@
 
 
 /datum/symptom/wizarditis/proc/teleport(datum/disease/advance/A)
-	var/turf/L = get_safe_random_station_turfs()
-	A.affected_mob.say("SCYAR NILA!")
-	do_teleport(A.affected_mob, L, channel = TELEPORT_CHANNEL_MAGIC)
-	playsound(get_turf(A.affected_mob), 'sound/weapons/zapbang.ogg', 50,1)
+	var/list/theareas = get_areas_in_range(80, A.affected_mob)
+	for(var/area/space/S in theareas)
+		theareas -= S
+
+	if(!theareas||!theareas.len)
+		return
+
+	var/area/thearea = pick(theareas)
+
+	var/list/L = list()
+	for(var/turf/T in get_area_turfs(thearea.type))
+		if(T.get_virtual_z_level() != A.affected_mob.get_virtual_z_level())
+			continue
+		if(T.name == "space")
```

Comment:
I'd check for the type of tile space is rather than it's name.

Reply:
This would make sense, yes. 

I just copy/pasted the code from wizarditis (the disease) to wizarditis (the symptom) and then corrected variable paths where needed

Reply:
Wizarditis is ancient and probably needs a refactor itself ;w;